### ISSUE TITLE
The sample size of train and test sets when using Runner( )

### ISSUE THREAD

**Issue by WangYunHong98** [amarquand/PCNtoolkit#349]
Hi all,

Thank you for your excellent tookit.

I have a question about using `Runner()` to implement k-fold validation. The arguments were 'cross_validate = True' and 'cv_folds = 5'.

My entire dataset comprises 1,453 observations and uses the `BLR()` estimate function. When I used `model.fit_prediction(train, test)`, the final outputs of the Z_train and Z_test samples summed to 1453. However, when I used `runner.fit_prediction(model, train, test)`, the sum of each fold of the `Z_train` and `Z_test` samples was 1162, indicating a lack of one fold sample (290.6). 

Why does this happen? Theoretically, the result of 5-fold cross-validation should be 1453 samples from Z_train plus Z_test.

By the way, my goal is to predict cognitive scores using deviation-Z through machine learning methods. When performing normative modeling in the previous step, should I treat gender as a covariate or a batch effect?

Looking forward you reply,
Yun-Hong

**Comment by WangYunHong98** [comment]
Thank you for your quick reply Stijn.

My understanding is that using the method you mentioned `runner.fit_predict(all_data)` is equivalent to performing a new analysis within Runner. Therefore, the test set for each fold may not correspond to the original test set obtained via `train, test = norm_data.train_test_split()`.

Below was my modeling code:
`covar_columns = ["age", "gender", "mean_FDs"] 
template_blr = BLR(
    name="template",
    basis_function_mean=BsplineBasisFunction(
        degree=3, nknots=5,
        basis_column=[0, 2]
    ),  
    fixed_effect=True,  
    heteroskedastic=True, 
    warp_name="warpsinharcsinh"
)`

I did not set the `batch effects` and took age, gender and mean framewise displacements into account (because the brain feature was functional connectivity), although I am not sure if this is correct.

Best,
Yun-Hong
